compiler/src/dotty/tools/dotc/classpath/AggregateClassPath.scala

```diff
@@ -18,33 +18,32 @@ import dotty.tools.io.{ AbstractFile, ClassPath, ClassRepresentation }
  * @param aggregates classpath instances containing entries which this class processes
  */
 case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
+
+  private[this] val packageIndex: collection.mutable.Map[String, Seq[ClassPath]] = collection.mutable.Map()
+  private def aggregatesForPackage(pkg: String): Seq[ClassPath] = packageIndex.synchronized {
+    packageIndex.getOrElseUpdate(pkg, aggregates.filter(_.hasPackage(pkg)))
+  }
+
   override def findClassFile(className: String): Option[AbstractFile] = {
```

Comment:
In the scalac implementation, this method is above `packageIndex`, it'd be good to keep it that way here to, to reduce the source diff between the two implementations.